diff --git a/../../SDL_original_2301/src/core/android/SDL_android.h b/../../SDL/src/core/android/SDL_android.h
index 20d1fc521..2d8d0faf6 100644
--- a/../../SDL_original_2301/src/core/android/SDL_android.h
+++ b/../../SDL/src/core/android/SDL_android.h
@@ -91,6 +91,7 @@ void Android_JNI_PollInputDevices(void);
 /* Haptic support */
 void Android_JNI_PollHapticDevices(void);
 void Android_JNI_HapticRun(int device_id, float intensity, int length);
+void Android_JNI_HapticRumble(int device_id, float low_frequency_intensity, float high_frequency_intensity, int length);
 void Android_JNI_HapticStop(int device_id);
 
 /* Video */
