diff --git a/D:/RotatingartLauncher/app/src/main/cpp/SDL_original_2301/android-project/app/src/main/java/org/libsdl/app/SDL.java b/D:/RotatingartLauncher/app/src/main/java/org/libsdl/app/SDL.java
index 44c21c1c7..0ec24203c 100644
--- a/D:/RotatingartLauncher/app/src/main/cpp/SDL_original_2301/android-project/app/src/main/java/org/libsdl/app/SDL.java
+++ b/D:/RotatingartLauncher/app/src/main/java/org/libsdl/app/SDL.java
@@ -49,10 +49,8 @@ public class SDL {
             // trip a bug in Android's native library loader which ReLinker works around.  (If
             // loadLibrary works properly, ReLinker will simply use the normal Android method
             // internally.)
-            //
             // To use ReLinker, just add it as a dependency.  For more information, see 
             // https://github.com/KeepSafe/ReLinker for ReLinker's repository.
-            //
             Class<?> relinkClass = mContext.getClassLoader().loadClass("com.getkeepsafe.relinker.ReLinker");
             Class<?> relinkListenerClass = mContext.getClassLoader().loadClass("com.getkeepsafe.relinker.ReLinker$LoadListener");
             Class<?> contextClass = mContext.getClassLoader().loadClass("android.content.Context");
